### PR TITLE
sending to federation address pr comments addressed

### PR DESCRIPTION
**Pull request by acharb**
from https://github.com/stellar/account-viewer-next/pull/48

### CODE REVIEW

File: src/components/SendTransaction/CreateTransaction.tsx [hunk 1]
```diff
@@ -63,6 +62,8 @@ export const CreateTransaction = (props: CreateProps) => {
         onInput({
           ...formData,
           federationAddress: response.account_id,
+          memoType: (response.memo_type as MemoType) || formData.memoType,
```
Comment (acharb) [inline src/components/SendTransaction/CreateTransaction.tsx:65]: @quietbits do you think it's safe to assume a correct memo type will be returned? I was also thinking of adding a check in "createMemo" that will default to MemoNone if an incorrect type given

Comment (quietbits) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: Yes, I don't think we need to check the returned memo type.

Comment (morleyzhi) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: I think if the federation server provides a memo, you have to use that memo; so if that's the case, I wouldn't override it with a user-provided one, and I wouldn't ask for one at all for starters. (I'm not 100% sure that's what's going on, it's just what this line reads like.)

Comment (quietbits) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: We can disable the memo type selection if it is returned. In the case where we have `memo` but no `memoType`, should we disable user selection as well?

Comment (acharb) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: Gotchya. Ya maybe I could make it easier to read. Basically it's overwriting any user-input Memo (or the default of empty) if one is returned from the federation server.

If a memo is returned and it has to be used, a user shouldn't be able to edit it before sending? Ie it should be locked? Or is it enough to not show the memo inputs

Comment (acharb) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: possibly a product question. CC @brunomuler @vcarl 

wondering about the case of a Memo being returned from a federation address request.

Should the memo be read-only for the user? On top of that, the edge case of a "memo" returned but no "memoType". Default the MemoType to text?

Comment (morleyzhi) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: I don't think the memos contain proprietary information. I would show and disable them if the anchor provides one.

If they don't provide one, it's safe to ask the user for one.

If they have a memo but not a memoType, that's pretty bad, though I can't tell if it's against the spec. My guess is it is. In that case I think you should set the type to "text." But I would try to look up the SEP that describes fed servers, if there is one, and see what it says.

Comment (morleyzhi) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0002.md

Comment (acharb) [reply src/components/SendTransaction/CreateTransaction.tsx:68]: Got it, thanks!

Comment (acharb) [reply src/components/SendTransaction/CreateTransaction.tsx]: Changes pushed up

**Pull request merged**